common4j/src/main/com/microsoft/identity/common/java/net/SSLSocketFactoryWrapper.java

```diff
@@ -0,0 +1,203 @@
+            sslSocket.setEnabledCipherSuites(getEnabledCipherSuites(sslSocket));
+            sslSocket.addHandshakeCompletedListener(new HandshakeCompletedListener() {
+                @Override
+                public void handshakeCompleted(final HandshakeCompletedEvent event) {
+                    setLastHandshakeTLSversion(event.getSession().getProtocol());
+                }
+            });
+        }
+        return socket;
+    }
+
+    /**
+     * Returns a list of protocols that are both supported by us {@see SUPPORTED_SSL_PROTOCOLS},
+     * and by the connection itself.
+     * */
+    private String[] getEnabledProtocols(@NonNull SSLSocket sslSocket) {
+        final List<String> enabledProtocols = new ArrayList<>();
+
+        final List<String> supportedProtocols = Arrays.asList(sslSocket.getSupportedProtocols());
+        for (final String protocol: SUPPORTED_SSL_PROTOCOLS){
+            if (supportedProtocols.contains(protocol)){
+                enabledProtocols.add(protocol);
+            }
+        }
+
+        final String[] array = new String[enabledProtocols.size()];
+        enabledProtocols.toArray(array);
+        return array;
+    }
+
+    /**
+     * Add required cipher for TLS 1.3
+     * See: https://datatracker.ietf.org/doc/html/rfc8446#section-9.1
+     */
+    private String[] getEnabledCipherSuites(@NonNull SSLSocket sslSocket) {
+        final List<String> enabledCipherSuites = new ArrayList<>();
+        Collections.addAll(enabledCipherSuites, sslSocket.getEnabledCipherSuites());
+
+        final List<String> supportedCipherSuites = Arrays.asList(sslSocket.getSupportedCipherSuites());
+        if (supportedCipherSuites.contains(TLS_AES_128_GCM_SHA256)) {
+            enabledCipherSuites.add(TLS_AES_128_GCM_SHA256);
+        }
+        if (supportedCipherSuites.contains(TLS_AES_256_GCM_SHA384)) {
+            enabledCipherSuites.add(TLS_AES_256_GCM_SHA384);
+        }
+        if (supportedCipherSuites.contains(TLS_CHACHA20_POLY1305_SHA256)) {
+            enabledCipherSuites.add(TLS_CHACHA20_POLY1305_SHA256);
+        }
+
+        final String[] array = new String[enabledCipherSuites.size()];
+        enabledCipherSuites.toArray(array);
+        return array;
+    }
+
+    /**
+     * Returns a {@link SSLSocketFactoryWrapper}.
+     */
+    @NonNull
+    public static synchronized SSLSocketFactory getSocketFactory(@Nullable SSLContext context) {
+        final String methodName = "getSocketFactory";
+        Logger.verbose(TAG + methodName, "getting SSLSocketFactory.");
+
+        if (context == null){
+            return sDefault;
+        }
+
+        final SocketFactory factory = context.getSocketFactory();
+        if (factory == null){
+            Logger.warn(TAG + methodName, "Failed to construct a SSLSocketFactory from SSLContext, returns the default one.");
+            return sDefault;
+        }
+
+        return new SSLSocketFactoryWrapper((SSLSocketFactory) context.getSocketFactory());
```

Comment:
```suggestion
        return new SSLSocketFactoryWrapper(context.getSocketFactory());
```

No need to cast since context.getSocketFactory() returns an SSLSocketFactory object.